### PR TITLE
feat: support native lp tokens refactor

### PR DESCRIPTION
**Pull request by ApolloGie**

### CODE REVIEW

File: cw-dex-astroport/tests/astroport_tests.rs [hunk 1]
```diff
@@ -141,48 +196,68 @@ mod tests {
...
         };
         let expected_out: AssetList = wasm.query(&contract_addr, &simulate_query).unwrap();
 
         // Withdraw liquidity with min_out one more than expected_out. Should fail.
-        let unwrap = Unwrap::Err("but expected");
+        let error_msg = match &pool_type {
+            PairType::Custom(t) if t == "concentrated" => {
+                "Min out is not supported for concentrated liquidity pools"
+            }
+            _ => "but expected",
```
Comment (pacmanifold) [inline cw-dex-astroport/tests/astroport_tests.rs:258]: why only part of expected message?

File: cw-dex-astroport/tests/astroport_tests.rs [hunk 2]
```diff
@@ -193,18 +268,25 @@ mod tests {
...
-            amount: contract_lp_token_balance,
+            amount: withdraw_amount,
             min_out,
         };
         unwrap.unwrap(runner.execute_cosmos_msgs::<MsgExecuteContractResponse>(
             &[withdraw_msg.into_cosmos_msg(contract_addr.clone(), vec![])],
             admin,
         ));
 
+        let _pool_res: PoolResponse = wasm.query(&pair_addr, &PairQueryMsg::Pool {}).unwrap();
```
Comment (pacmanifold) [inline cw-dex-astroport/tests/astroport_tests.rs:279]: why query value if never used?

File: cw-dex-astroport/tests/astroport_tests.rs [hunk 3]
```diff
@@ -214,29 +296,49 @@ mod tests {
...
+                &[withdraw_msg.into_cosmos_msg(contract_addr.clone(), vec![])],
+                admin,
+            )
+            .unwrap();
+
+        // Query LP token balance after
+        let lp_token_balance_after = query_asset_balance(&runner, &lp_token, &contract_addr);
+
+        // Assert that LP token balance is zero after withdrawing all liquidity
+        assert_eq!(lp_token_balance_after, Uint128::zero());
```
Comment (pacmanifold) [inline cw-dex-astroport/tests/astroport_tests.rs:330]: why add this?

File: cw-dex-astroport/tests/astroport_tests.rs [hunk 4]
```diff
@@ -214,29 +296,49 @@ mod tests {
...
         runner: &'a R,
         contract_addr: String,
-        lp_token_addr: String,
+        lp_token_denom: String,
         signer: &SigningAccount,
     ) -> ExecuteResponse<MsgExecuteContractResponse> {
         // Query LP token balance
         let lp_token_balance =
-            cw20_balance_query(runner, lp_token_addr, contract_addr.clone()).unwrap();
+            bank_balance_query(runner, signer.address().clone(), lp_token_denom.clone()).unwrap();
```
Comment (pacmanifold) [inline cw-dex-astroport/tests/astroport_tests.rs:341]: this assumes lp token is a native denom. which is not done in the rest of the code base. if this is intentional then let's rename the function to `stake_all_native_lp_tokens`. 

but i think we should either have tests for both native and cw20 though or remove cw20 support

File: cw-dex-astroport/tests/astroport_tests.rs [hunk 5]
```diff
@@ -266,29 +371,23 @@ mod tests {
     ) -> RunnerResult<()> {
```
Comment (pacmanifold) [inline cw-dex-astroport/tests/astroport_tests.rs:371]: rename test to `test_stake_and_unstake_native_lp` since it only tests the native case

**Review by pacmanifold — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-format.yml [hunk 6]
```diff
@@ -25,6 +25,9 @@ jobs:
       - name: Install nightly toolchain
         run: cargo make install-nightly
 
+      - name: Install stable toolchain
+        run: cargo make install-stable
+
```
Comment (pacmanifold) [inline .github/workflows/lint-format.yml:30]: if we add this we should remove nightly?

File: cw-dex-astroport/tests/astroport_tests.rs [hunk 7]
```diff
@@ -52,44 +80,60 @@ mod tests {
...
+                pool_type.clone(),
+                use_liquidity_manager,
+                initial_liquidity,
+            )
+            .unwrap();
         let admin = &accs[0];
         let wasm = Wasm::new(&runner);
+        let _pair_config_res: PairInfo = wasm.query(&pair_addr, &PairQueryMsg::Pair {}).unwrap();
 
+        let lp_token = if lp_token.starts_with(admin.prefix()) || lp_token.starts_with("contract") {
```
[the remark this answers is not in the document]
Comment (apollo-sturdy) [reply cw-dex-astroport/tests/astroport_tests.rs:119]: its for test-tube and multi-test respectively

File: cw-dex-astroport/tests/astroport_tests.rs (hunk 1 above)
Comment (apollo-sturdy) [reply cw-dex-astroport/tests/astroport_tests.rs:258]: idk ask the person who wrote that (probably me 2 years ago)

File: cw-dex-astroport/tests/astroport_tests.rs (hunk 2 above)
Comment (apollo-sturdy) [reply cw-dex-astroport/tests/astroport_tests.rs:279]: remove it then

File: cw-dex-astroport/tests/astroport_tests.rs (hunk 4 above)
Comment (apollo-sturdy) [reply cw-dex-astroport/tests/astroport_tests.rs:341]: just change it to query lp_token_balance there's a helper

File: .github/workflows/lint-format.yml (hunk 6 above)
Comment (apollo-sturdy) [reply .github/workflows/lint-format.yml:30]: idk do we use both or nah? check the makefile

Comment (pacmanifold) [reply .github/workflows/lint-format.yml:30]: we need both. some features we use for format are not in stable 1.79

**Review by apollo-sturdy — APPROVED** [review]
lgtm

**Pull request merged**